### PR TITLE
RESTWS-567 Added ConceptStopWord REST URL's

### PR DESCRIPTION
**Pull request by tmarzeion**
Jira ticket:
https://issues.openmrs.org/browse/RESTWS-567

### CODE REVIEW

File: omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/ConceptStopwordResource1_9.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+ */
+@Resource(name = RestConstants.VERSION_1 + "/conceptstopword", supportedClass = ConceptStopWord.class, supportedOpenmrsVersions = {"1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*"})
+public class ConceptStopwordResource1_9 extends DelegatingCrudResource<ConceptStopWord> {
+
+    /**
+     * @see DelegatingCrudResource#getRepresentationDescription(Representation)
+     */
+    @Override
+    public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+        if (rep instanceof RefRepresentation || rep instanceof DefaultRepresentation) {
```
Comment (rkorytkowski) [inline omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/ConceptStopwordResource1_9.java]: ref representation should contain only the following properties: uuid, display, retired (if set to true), selflink

File: omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/ConceptStopwordResource1_9.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            description.addProperty("uuid");
+            description.addProperty("value");
+            description.addProperty("locale");
+            description.addSelfLink();
+            return description;
+        }
+        return null;
+    }
+
+    @PropertyGetter("value")
```
Comment (rkorytkowski) [inline omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/ConceptStopwordResource1_9.java]: It should be not value, but "display"

File: omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/ConceptStopwordResource1_9.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+     *      org.openmrs.module.webservices.rest.web.RequestContext)
+     */
+    @Override
+    public void purge(ConceptStopWord delegate, RequestContext context) throws ResponseException {
+        throw new ResourceDoesNotSupportOperationException();
+    }
+
+    @Override
+    protected void delete(ConceptStopWord delegate, String reason, RequestContext context) throws ResponseException {
+        Context.getConceptService().deleteConceptStopWord(delegate.getId());
```
Comment (rkorytkowski) [inline omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/ConceptStopwordResource1_9.java]: I think deleteConceptStopWord is a purge operation and not retiring. I'm sorry about the inconsistency. Please use deleteConceptStopWord in purge and throw ResourceDoesNotSupportOperationException here.

Comment (tmarzeion) [inline omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/ConceptStopwordResource1_9.java]: Deleting doesn't work and are throwing responses with ResourceDoesNotSupportOperationException.
Thats why I "switched" delete methods between purge and delete.

File: omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/ConceptStopwordResource1_9.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("display");
+            description.addSelfLink();
+            return description;
+        } else if (rep instanceof DefaultRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("display");
+            description.addProperty("locale");
```
Comment (rkorytkowski) [inline omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/ConceptStopwordResource1_9.java]: "value" property must be added as well.

File: omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/ConceptStopwordResource1_9.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+            description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+            description.addSelfLink();
+            return description;
+        } else if (rep instanceof FullRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("display");
+            description.addProperty("locale");
+            description.addSelfLink();
```
Comment (rkorytkowski) [inline omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/ConceptStopwordResource1_9.java]: "value" property must be added as well.

**Pull request merged**